RestaurantReservationSystem/Program.cs

```diff
@@ -1 +1,59 @@
-﻿
\ No newline at end of file
+﻿using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using RestaurantReservation.Db;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using RestaurantReservation.Services;
+using RestaurantReservation.Db.Repositories;
+
+var host = Host.CreateDefaultBuilder()
+    .ConfigureLogging(logging =>
+     {
+         logging.ClearProviders();
+     })
+    .ConfigureServices((context, services) =>
+    {
+        services.AddDbContext<RestaurantReservationDbContext>(options =>
+            options.UseSqlServer(context.Configuration.GetConnectionString("DefaultConnection")));
+
+        services.AddScoped<RestaurantOperations>();
+        services.AddScoped<RestaurantService>();
+
+        services.AddScoped<CustomerOperations>();
+        services.AddScoped<CustomerService>();
+
+        services.AddScoped<EmployeeOperations>();
+        services.AddScoped<EmployeeService>();
+
+        services.AddScoped<ReservationOperations>();
+        services.AddScoped<ReservationService>();
+
+        services.AddScoped<TableOperations>();
+        services.AddScoped<TableService>();
+
+        services.AddScoped<MenuItemOperations>();
+        services.AddScoped<MenuItemService>();
+
+        services.AddScoped<OrderOperations>();
+        services.AddScoped<OrderService>();
+
+        services.AddScoped<OrderItemOperations>();
+        services.AddScoped<OrderItemService>();
+
+        services.AddScoped<EmployeeRepository>();
+        services.AddScoped<ReservationRepository>();
+        services.AddScoped<OrderRepository>();
+        services.AddScoped<MenuItemRepository>();
+
+        services.AddScoped<ReservationReportRepository>();
+        services.AddScoped<EmployeeReportRepository>();
```

Comment:
Lets also create interfaces for each service and repo. This will give you a good practice for the upcoming projects, and inject interfaces in the constructors instead.

Interfaces are important to be able to apply the dependency inversion principle, we want to make sure we only expose the interfaces and make them public and make the concrete classes private. So other projects will be seeing the public interfaces from the db layer and since repos classes are private, no one will be able to see them from the outside.

Later on, you will learn more about layers architecture and clean architecture, where we will be introducing DB, Domain and App layers.